Installation failed with iojs v2.0.0

```
> socketwatcher@0.2.1 install /root/zpn-deploy/source/node_modules/pcappp/node_modules/pcap/node_modules/socketwatcher
> node-gyp rebuild

make: Entering directory `/root/zpn-deploy/source/node_modules/pcappp/node_modules/pcap/node_modules/socketwatcher/build'
  CXX(target) Release/obj.target/socketwatcher/socket_watcher.o
In file included from ../node_modules/nan/nan_new.h:190:0,
                 from ../node_modules/nan/nan.h:80,
                 from ../socket_watcher.hpp:8,
                 from ../socket_watcher.cpp:5:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static NanIntern::FactoryBase<v8::Signature>::return_t NanIntern::Factory<v8::Signature>::New(NanIntern::Factory<v8::Signature>::FTH, int, NanIntern::Factory<v8::Signature>::FTH*)’:
../node_modules/nan/nan_implementation_12_inl.h:181:76: error: no matching function for call to ‘v8::Signature::New(v8::Isolate*, NanIntern::Factory<v8::Signature>::FTH&, int&, NanIntern::Factory<v8::Signature>::FTH*&)’
   return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
                                                                            ^
../node_modules/nan/nan_implementation_12_inl.h:181:76: note: candidate is:
In file included from /root/.node-gyp/2.0.0/src/node.h:42:0,
                 from ../node_modules/nan/nan.h:24,
                 from ../socket_watcher.hpp:8,
                 from ../socket_watcher.cpp:5:
/root/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note: static v8::Local<v8::Signature> v8::Signature::New(v8::Isolate*, v8::Handle<v8::FunctionTemplate>)
   static Local<Signature> New(
                           ^
/root/.node-gyp/2.0.0/deps/v8/include/v8.h:4188:27: note:   candidate expects 2 arguments, 4 provided
make: *** [Release/obj.target/socketwatcher/socket_watcher.o] Error 1
make: Leaving directory `/root/zpn-deploy/source/node_modules/pcappp/node_modules/pcap/node_modules/socketwatcher/build'
gyp ERR! build error
```

Works find with iojs v1.8.1 :)

I can confirm:
Does not work on iojs-2.2.1:
```
> socketwatcher@0.2.1 install /Users/thomas/...../node_modules/pcap/node_modules/socketwatcher
> node-gyp rebuild

  CXX(target) Release/obj.target/socketwatcher/socket_watcher.o
In file included from ../socket_watcher.cpp:5:
In file included from ../socket_watcher.hpp:8:
In file included from ../node_modules/nan/nan.h:80:
In file included from ../node_modules/nan/nan_new.h:190:
../node_modules/nan/nan_implementation_12_inl.h:181:66: error: too many arguments to function call, expected at most 2, have 4
  return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
         ~~~~~~~~~~~~~~~~~~                                      ^~~~~~~~~~
/Users/thomas/.node-gyp/2.2.1/deps/v8/include/v8.h:4188:3: note: 'New' declared here
  static Local<Signature> New(
  ^
1 error generated.
make: *** [Release/obj.target/socketwatcher/socket_watcher.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/thomas/.nvm/versions/io.js/v2.2.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:199:12)
gyp ERR! System Darwin 14.4.0
gyp ERR! command "/Users/thomas/.nvm/versions/io.js/v2.2.1/bin/iojs" "/Users/thomas/.nvm/versions/io.js/v2.2.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/thomas/...../node_modules/pcap/node_modules/socketwatcher
gyp ERR! node -v v2.2.1
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok
```
Does work on iojs-1.8.2.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,23 @@
 language: node_js
 node_js:
+  - 4
+  - 3
+  - 2
+  - 1
   - "0.12"
   - "0.11"
   - "0.10"
-  - iojs
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - gcc-4.9
+      - g++-4.9
 before_script:
   - npm install -g grunt-cli
 before_install:
+  - export CC="gcc-4.9" CXX="g++-4.9"
   - sudo apt-get install libpcap-dev
 script:
   - grunt travis

--- a/package.json
+++ b/package.json
@@ -11,9 +11,12 @@
     "type": "git",
     "url": "git://github.com/mranney/node_pcap.git"
   },
+  "engines": {
+    "node": ">=0.10"
+  },
   "dependencies": {
-    "nan": "~1.6.1",
-    "socketwatcher": "git://github.com/zerouid/node-socketwatcher.git"
+    "nan": "^2.0.9",
+    "socketwatcher": "^0.3.0"
   },
   "devDependencies": {
     "grunt": "^0.4.5",

--- a/pcap_binding.cc
+++ b/pcap_binding.cc
@@ -12,7 +12,6 @@
 #include "pcap_session.h"
 
 using namespace v8;
-using namespace node;
 
 // Helper method, convert a sockaddr* (AF_INET or AF_INET6) to a string, and set it as the property
 // named 'key' in the Address object you pass in.
@@ -31,64 +30,65 @@ void SetAddrStringHelper(const char* key, sockaddr *addr, Local<Object> Address)
       size = INET6_ADDRSTRLEN;
     }
     const char* address = inet_ntop(addr->sa_family, src, dst_addr, size);
-    Address->Set(NanNew(key), NanNew(address));
+    Address->Set(Nan::New(key).ToLocalChecked(), Nan::New(address).ToLocalChecked());
   }
 }
 
 
 NAN_METHOD(FindAllDevs)
 {
-    NanScope();
+    Nan::HandleScope scope;
+
     char errbuf[PCAP_ERRBUF_SIZE];
     pcap_if_t *alldevs, *cur_dev;
 
     if (pcap_findalldevs(&alldevs, errbuf) == -1 || alldevs == NULL) {
-        NanThrowTypeError(errbuf);
-        NanReturnUndefined();
+        Nan::ThrowTypeError(errbuf);
+        return;
     }
 
-    Local<Array> DevsArray = NanNew<Array>();
+    Local<Array> DevsArray = Nan::New<Array>();
 
     int i = 0;
     for (cur_dev = alldevs ; cur_dev != NULL ; cur_dev = cur_dev->next, i++) {
-        Local<Object> Dev = NanNew<Object>();
+        Local<Object> Dev = Nan::New<Object>();
 
-        Dev->Set(NanNew("name"), NanNew(cur_dev->name));
+        Dev->Set(Nan::New("name").ToLocalChecked(), Nan::New(cur_dev->name).ToLocalChecked());
         if (cur_dev->description != NULL) {
-            Dev->Set(NanNew("description"), NanNew(cur_dev->description));
+            Dev->Set(Nan::New("description").ToLocalChecked(), Nan::New(cur_dev->description).ToLocalChecked());
         }
-        Local<Array> AddrArray = NanNew<Array>();
+        Local<Array> AddrArray = Nan::New<Array>();
         int j = 0;
         for (pcap_addr_t *cur_addr = cur_dev->addresses ; cur_addr != NULL ; cur_addr = cur_addr->next, j++) {
           if (cur_addr->addr){
               int af = cur_addr->addr->sa_family;
               if(af == AF_INET || af == AF_INET6){
-                Local<Object> Address = NanNew<Object>();
+                Local<Object> Address = Nan::New<Object>();
                 SetAddrStringHelper("addr", cur_addr->addr, Address);
                 SetAddrStringHelper("netmask", cur_addr->netmask, Address);
                 SetAddrStringHelper("broadaddr", cur_addr->broadaddr, Address);
                 SetAddrStringHelper("dstaddr", cur_addr->dstaddr, Address);
-                AddrArray->Set(NanNew<Integer>(j), Address);
+                AddrArray->Set(Nan::New<Integer>(j), Address);
               }
            }
         }
 
-        Dev->Set(NanNew("addresses"), AddrArray);
+        Dev->Set(Nan::New("addresses").ToLocalChecked(), AddrArray);
 
         if (cur_dev->flags & PCAP_IF_LOOPBACK) {
-            Dev->Set(NanNew("flags"), NanNew("PCAP_IF_LOOPBACK"));
+            Dev->Set(Nan::New("flags").ToLocalChecked(), Nan::New("PCAP_IF_LOOPBACK").ToLocalChecked());
         }
 
-        DevsArray->Set(NanNew<Integer>(i), Dev);
+        DevsArray->Set(Nan::New<Integer>(i), Dev);
     }
 
     pcap_freealldevs(alldevs);
-    NanReturnValue(DevsArray);
+    info.GetReturnValue().Set(DevsArray);
 }
 
 NAN_METHOD(DefaultDevice)
 {
-    NanScope();
+    Nan::HandleScope scope;
     char errbuf[PCAP_ERRBUF_SIZE];
 
     // Look up the first device with an address, pcap_lookupdev() just returns the first non-loopback device.
@@ -97,13 +97,13 @@ NAN_METHOD(DefaultDevice)
     bool found = false;
 
     if (pcap_findalldevs(&alldevs, errbuf) == -1) {
-      NanThrowError(errbuf);
-      NanReturnUndefined();
+      Nan::ThrowError(errbuf);
+      return;
     }
 
     if (alldevs == NULL) {
-      NanThrowError("pcap_findalldevs didn't find any devs");
-      NanReturnUndefined();
+      Nan::ThrowError("pcap_findalldevs didn't find any devs");
+      return;
     }
 
     for (dev = alldevs; dev != NULL; dev = dev->next) {
@@ -112,7 +112,7 @@ NAN_METHOD(DefaultDevice)
                 // TODO - include IPv6 addresses in DefaultDevice guess
                 // if (addr->addr->sa_family == AF_INET || addr->addr->sa_family == AF_INET6) {
                 if (addr->addr->sa_family == AF_INET) {
-                    NanReturnValue(NanNew(dev->name));
+                    info.GetReturnValue().Set(Nan::New(dev->name).ToLocalChecked());
                     found = true;
                     break;
                 }
@@ -125,25 +125,23 @@ NAN_METHOD(DefaultDevice)
     }
 
     pcap_freealldevs(alldevs);
-    NanReturnUndefined();
+    return;
 }
 
 NAN_METHOD(LibVersion)
 {
-    NanScope();
-
-    NanReturnValue(NanNew(pcap_lib_version()));
+    info.GetReturnValue().Set(Nan::New(pcap_lib_version()).ToLocalChecked());
 }
 
 void Initialize(Handle<Object> exports)
 {
-//    HandleScope scope;
+    Nan::HandleScope scope;
 
     PcapSession::Init(exports);
 
-    exports->Set(NanNew("findalldevs"), NanNew<FunctionTemplate>(FindAllDevs)->GetFunction());
-    exports->Set(NanNew("default_device"), NanNew<FunctionTemplate>(DefaultDevice)->GetFunction());
-    exports->Set(NanNew("lib_version"), NanNew<FunctionTemplate>(LibVersion)->GetFunction());
+    exports->Set(Nan::New("findalldevs").ToLocalChecked(), Nan::New<FunctionTemplate>(FindAllDevs)->GetFunction());
+    exports->Set(Nan::New("default_device").ToLocalChecked(), Nan::New<FunctionTemplate>(DefaultDevice)->GetFunction());
+    exports->Set(Nan::New("lib_version").ToLocalChecked(), Nan::New<FunctionTemplate>(LibVersion)->GetFunction());
 }
 
 NODE_MODULE(pcap_binding, Initialize)

--- a/pcap_session.cc
+++ b/pcap_session.cc
@@ -8,43 +8,42 @@
 
 using namespace v8;
 
-Persistent<Function> PcapSession::constructor;
+Nan::Persistent<Function> PcapSession::constructor;
 
 PcapSession::PcapSession() {};
 PcapSession::~PcapSession() {};
 
 void PcapSession::Init(Handle<Object> exports) {
-  NanScope();
+  Nan::HandleScope scope;
   // Prepare constructor template
-  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
-  tpl->SetClassName(NanNew("PcapSession"));
+  Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(New);
+  tpl->SetClassName(Nan::New("PcapSession").ToLocalChecked());
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
 
   // Prototype
-  NODE_SET_PROTOTYPE_METHOD(tpl, "open_live", OpenLive);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "open_offline", OpenOffline);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "dispatch", Dispatch);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "fileno", Fileno);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "close", Close);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "stats", Stats);
-  NODE_SET_PROTOTYPE_METHOD(tpl, "inject", Inject);
+  Nan::SetPrototypeMethod(tpl, "open_live", OpenLive);
+  Nan::SetPrototypeMethod(tpl, "open_offline", OpenOffline);
+  Nan::SetPrototypeMethod(tpl, "dispatch", Dispatch);
+  Nan::SetPrototypeMethod(tpl, "fileno", Fileno);
+  Nan::SetPrototypeMethod(tpl, "close", Close);
+  Nan::SetPrototypeMethod(tpl, "stats", Stats);
+  Nan::SetPrototypeMethod(tpl, "inject", Inject);
 
-  NanAssignPersistent(constructor, tpl->GetFunction());
-  exports->Set(NanNew("PcapSession"), tpl->GetFunction());
+  constructor.Reset(tpl->GetFunction());
+  exports->Set(Nan::New("PcapSession").ToLocalChecked(), tpl->GetFunction());
 }
 
-NAN_METHOD(PcapSession::New) {
-  NanScope();
-
-  if (args.IsConstructCall()) {
+void PcapSession::New(const Nan::FunctionCallbackInfo<Value>& info) {
+  Nan::HandleScope scope;
+  if (info.IsConstructCall()) {
     // Invoked as constructor: `new MyObject(...)`
     PcapSession* obj = new PcapSession();
-    obj->Wrap(args.This());
-    NanReturnValue(args.This());
+    obj->Wrap(info.This());
+    info.GetReturnValue().Set(info.This());
   } else {
     // Invoked as plain function `MyObject(...)`, turn into construct call.
-    Local<Function> cons = NanNew<Function>(constructor);
-    NanReturnValue(cons->NewInstance());
+    Local<Function> cons = Nan::New<Function>(constructor);
+    info.GetReturnValue().Set(cons->NewInstance());
   }
 }
 
@@ -61,7 +60,7 @@ NAN_METHOD(PcapSession::New) {
 // 6. session.dispatch callback (pcap.js)
 //
 void PcapSession::PacketReady(u_char *s, const struct pcap_pkthdr* pkthdr, const u_char* packet) {
-    NanScope();
+    Nan::HandleScope scope;
 
     PcapSession* session = (PcapSession *)s;
 
@@ -81,41 +80,40 @@ void PcapSession::PacketReady(u_char *s, const struct pcap_pkthdr* pkthdr, const
     memcpy(session->header_data + 8, &(pkthdr->caplen), 4);
     memcpy(session->header_data + 12, &(pkthdr->len), 4);
 
+    Nan::TryCatch try_catch;
 
-    TryCatch try_catch;
-
-    NanMakeCallback(NanGetCurrentContext()->Global(), NanNew(session->packet_ready_cb), 0, NULL);
+    Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(session->packet_ready_cb), 0, NULL);
 
     if (try_catch.HasCaught())  {
-        node::FatalException(try_catch);
+        Nan::FatalException(try_catch);
     }
 }
 
-NAN_METHOD(PcapSession::Dispatch)
+void PcapSession::Dispatch(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
 
-    if (args.Length() != 2) {
-        NanThrowTypeError("Dispatch takes exactly two arguments");
-        NanReturnUndefined();
+    if (info.Length() != 2) {
+        Nan::ThrowTypeError("Dispatch takes exactly two arguments");
+        return;
     }
 
-    if (!node::Buffer::HasInstance(args[0])) {
-        NanThrowTypeError("First argument must be a buffer");
-        NanReturnUndefined();
+    if (!node::Buffer::HasInstance(info[0])) {
+        Nan::ThrowTypeError("First argument must be a buffer");
+        return;
     }
 
-    if (!node::Buffer::HasInstance(args[1])) {
-        NanThrowTypeError("Second argument must be a buffer");
-        NanReturnUndefined();
+    if (!node::Buffer::HasInstance(info[1])) {
+        Nan::ThrowTypeError("Second argument must be a buffer");
+        return;
     }
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.This());
 
-    Local<Object> buffer_obj = args[0]->ToObject();
+    Local<Object> buffer_obj = info[0]->ToObject();
     session->buffer_data = node::Buffer::Data(buffer_obj);
     session->buffer_length = node::Buffer::Length(buffer_obj);
-    Local<Object> header_obj = args[1]->ToObject();
+    Local<Object> header_obj = info[1]->ToObject();
     session->header_data = node::Buffer::Data(header_obj);
 
     int packet_count;
@@ -123,51 +121,51 @@ NAN_METHOD(PcapSession::Dispatch)
         packet_count = pcap_dispatch(session->pcap_handle, 1, PacketReady, (u_char *)session);
     } while (packet_count > 0);
 
-    NanReturnValue(NanNew<Integer>(packet_count));
+    info.GetReturnValue().Set(Nan::New<Integer>(packet_count));
 }
 
-_NAN_METHOD_RETURN_TYPE PcapSession::Open(bool live, _NAN_METHOD_ARGS)
+void PcapSession::Open(bool live, const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
     char errbuf[PCAP_ERRBUF_SIZE];
 
-    if (args.Length() == 6) {
-        if (!args[0]->IsString()) {
-            NanThrowTypeError("pcap Open: args[0] must be a String");
-            NanReturnUndefined();
+    if (info.Length() == 6) {
+        if (!info[0]->IsString()) {
+            Nan::ThrowTypeError("pcap Open: info[0] must be a String");
+            return;
         }
-        if (!args[1]->IsString()) {
-            NanThrowTypeError("pcap Open: args[1] must be a String");
-            NanReturnUndefined();
+        if (!info[1]->IsString()) {
+            Nan::ThrowTypeError("pcap Open: info[1] must be a String");
+            return;
         }
-        if (!args[2]->IsInt32()) {
-            NanThrowTypeError("pcap Open: args[2] must be a Number");
-            NanReturnUndefined();
+        if (!info[2]->IsInt32()) {
+            Nan::ThrowTypeError("pcap Open: info[2] must be a Number");
+            return;
         }
-        if (!args[3]->IsString()) {
-            NanThrowTypeError("pcap Open: args[3] must be a String");
-            NanReturnUndefined();
+        if (!info[3]->IsString()) {
+            Nan::ThrowTypeError("pcap Open: info[3] must be a String");
+            return;
         }
-        if (!args[4]->IsFunction()) {
-            NanThrowTypeError("pcap Open: args[4] must be a Function");
-            NanReturnUndefined();
+        if (!info[4]->IsFunction()) {
+            Nan::ThrowTypeError("pcap Open: info[4] must be a Function");
+            return;
         }
-        if (!args[5]->IsBoolean()) {
-            NanThrowTypeError("pcap Open: args[5] must be a Boolean");
-            NanReturnUndefined();
+        if (!info[5]->IsBoolean()) {
+            Nan::ThrowTypeError("pcap Open: info[5] must be a Boolean");
+            return;
         }
     } else {
-        NanThrowTypeError("pcap Open: expecting 6 arguments");
-        NanReturnUndefined();
+        Nan::ThrowTypeError("pcap Open: expecting 6 arguments");
+        return;
     }
-    NanUtf8String device(args[0]->ToString());
-    NanUtf8String filter(args[1]->ToString());
-    int buffer_size = args[2]->Int32Value();
-    NanUtf8String pcap_output_filename(args[3]->ToString());
+    Nan::Utf8String device(info[0]->ToString());
+    Nan::Utf8String filter(info[1]->ToString());
+    int buffer_size = info[2]->Int32Value();
+    Nan::Utf8String pcap_output_filename(info[3]->ToString());
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.This());
 
-    NanAssignPersistent(session->packet_ready_cb, args[4].As<Function>());
+    session->packet_ready_cb.Reset(info[4].As<Function>());
     session->pcap_dump_handle = NULL;
 
     if (live) {
@@ -179,79 +177,79 @@ _NAN_METHOD_RETURN_TYPE PcapSession::Open(bool live, _NAN_METHOD_ARGS)
 
         session->pcap_handle = pcap_create((char *) *device, errbuf);
         if (session->pcap_handle == NULL) {
-            NanThrowError(errbuf);
-            NanReturnUndefined();
+            Nan::ThrowError(errbuf);
+            return;
         }
 
         // 64KB is the max IPv4 packet size
         if (pcap_set_snaplen(session->pcap_handle, 65535) != 0) {
-            NanThrowError("error setting snaplen");
-            NanReturnUndefined();
+            Nan::ThrowError("error setting snaplen");
+            return;
         }
 
         // always use promiscuous mode
         if (pcap_set_promisc(session->pcap_handle, 1) != 0) {
-            NanThrowError("error setting promiscuous mode");
-            NanReturnUndefined();
+            Nan::ThrowError("error setting promiscuous mode");
+            return;
         }
 
         // Try to set buffer size.  Sometimes the OS has a lower limit that it will silently enforce.
         if (pcap_set_buffer_size(session->pcap_handle, buffer_size) != 0) {
-            NanThrowError("error setting buffer size");
-            NanReturnUndefined();
+            Nan::ThrowError("error setting buffer size");
+            return;
         }
 
         // set "timeout" on read, even though we are also setting nonblock below.  On Linux this is required.
         if (pcap_set_timeout(session->pcap_handle, 1000) != 0) {
-            NanThrowError("error setting read timeout");
-            NanReturnUndefined();
+            Nan::ThrowError("error setting read timeout");
+            return;
         }
 
         // fixes a previous to-do that was here.
-        if (args.Length() == 6) {
-            if (args[5]->Int32Value()) {
+        if (info.Length() == 6) {
+            if (info[5]->Int32Value()) {
                 if (pcap_set_rfmon(session->pcap_handle, 1) != 0) {
-                    NanThrowError(pcap_geterr(session->pcap_handle));
-                    NanReturnUndefined();
+                    Nan::ThrowError(pcap_geterr(session->pcap_handle));
+                    return;
                 }
             }
         }
 
         if (pcap_activate(session->pcap_handle) != 0) {
-            NanThrowError(pcap_geterr(session->pcap_handle));
-            NanReturnUndefined();
+            Nan::ThrowError(pcap_geterr(session->pcap_handle));
+            return;
         }
 
         if (strlen((char *) *pcap_output_filename) > 0) {
             session->pcap_dump_handle = pcap_dump_open(session->pcap_handle, (char *) *pcap_output_filename);
             if (session->pcap_dump_handle == NULL) {
-                NanThrowError("error opening dump");
-                NanReturnUndefined();
+                Nan::ThrowError("error opening dump");
+                return;
             }
         }
 
         if (pcap_setnonblock(session->pcap_handle, 1, errbuf) == -1) {
-          NanThrowError(errbuf);
-          NanReturnUndefined();
+          Nan::ThrowError(errbuf);
+          return;
         }
     } else {
         // Device is the path to the savefile
         session->pcap_handle = pcap_open_offline((char *) *device, errbuf);
         if (session->pcap_handle == NULL) {
-            NanThrowError(errbuf);
-            NanReturnUndefined();
+            Nan::ThrowError(errbuf);
+            return;
         }
     }
 
     if (filter.length() != 0) {
       if (pcap_compile(session->pcap_handle, &session->fp, (char *) *filter, 1, session->net) == -1) {
-        NanThrowError(pcap_geterr(session->pcap_handle));
-        NanReturnUndefined();
+        Nan::ThrowError(pcap_geterr(session->pcap_handle));
+        return;
       }
 
       if (pcap_setfilter(session->pcap_handle, &session->fp) == -1) {
-        NanThrowError(pcap_geterr(session->pcap_handle));
-        NanReturnUndefined();
+        Nan::ThrowError(pcap_geterr(session->pcap_handle));
+        return;
       }
       pcap_freecode(&session->fp);
     }
@@ -272,43 +270,43 @@ _NAN_METHOD_RETURN_TYPE PcapSession::Open(bool live, _NAN_METHOD_ARGS)
     Local<Value> ret;
     switch (link_type) {
     case DLT_NULL:
-        ret = NanNew("LINKTYPE_NULL");
+        ret = Nan::New("LINKTYPE_NULL").ToLocalChecked();
         break;
     case DLT_EN10MB: // most wifi interfaces pretend to be "ethernet"
-        ret =  NanNew("LINKTYPE_ETHERNET");
+        ret =  Nan::New("LINKTYPE_ETHERNET").ToLocalChecked();
         break;
     case DLT_IEEE802_11_RADIO: // 802.11 "monitor mode"
-        ret = NanNew("LINKTYPE_IEEE802_11_RADIO");
+        ret = Nan::New("LINKTYPE_IEEE802_11_RADIO").ToLocalChecked();
         break;
     case DLT_RAW: // "raw IP"
-        ret = NanNew("LINKTYPE_RAW");
+        ret = Nan::New("LINKTYPE_RAW").ToLocalChecked();
         break;
     case DLT_LINUX_SLL: // "Linux cooked capture"
-        ret = NanNew("LINKTYPE_LINUX_SLL");
+        ret = Nan::New("LINKTYPE_LINUX_SLL").ToLocalChecked();
         break;
     default:
         snprintf(errbuf, PCAP_ERRBUF_SIZE, "Unknown linktype %d", link_type);
-        ret = NanNew(errbuf);
+        ret = Nan::New(errbuf).ToLocalChecked();
         break;
     }
-    NanReturnValue(ret);
+    info.GetReturnValue().Set(ret);
 }
 
-NAN_METHOD(PcapSession::OpenLive)
+void PcapSession::OpenLive(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    return Open(true, args);
+    return Open(true, info);
 }
 
-NAN_METHOD(PcapSession::OpenOffline)
+void PcapSession::OpenOffline(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    return Open(false, args);
+    return Open(false, info);
 }
 
-NAN_METHOD(PcapSession::Close)
+void PcapSession::Close(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.Holder());
 
     if (session->pcap_dump_handle != NULL) {
         pcap_dump_close(session->pcap_dump_handle);
@@ -316,70 +314,69 @@ NAN_METHOD(PcapSession::Close)
     }
 
     pcap_close(session->pcap_handle);
-    NanDisposePersistent(session->packet_ready_cb);
-    NanReturnUndefined();
+    session->packet_ready_cb.Reset();
+    return;
 }
 
-NAN_METHOD(PcapSession::Fileno)
+void PcapSession::Fileno(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.Holder());
 
     int fd = pcap_get_selectable_fd(session->pcap_handle);
 
-    NanReturnValue(NanNew<Integer>(fd));
+    info.GetReturnValue().Set(Nan::New<Integer>(fd));
 }
 
-NAN_METHOD(PcapSession::Stats)
+void PcapSession::Stats(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
 
     struct pcap_stat ps;
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.Holder());
 
     if (pcap_stats(session->pcap_handle, &ps) == -1) {
-        NanThrowError("Error in pcap_stats");
-        NanReturnUndefined();
+        Nan::ThrowError("Error in pcap_stats");
+        return;
         // TODO - use pcap_geterr to figure out what the error was
     }
 
-    Local<Object> stats_obj = NanNew<Object>();
+    Local<Object> stats_obj = Nan::New<Object>();
 
-    stats_obj->Set(NanNew("ps_recv"), NanNew<Integer>(ps.ps_recv));
-    stats_obj->Set(NanNew("ps_drop"), NanNew<Integer>(ps.ps_drop));
-    stats_obj->Set(NanNew("ps_ifdrop"), NanNew<Integer>(ps.ps_ifdrop));
+    stats_obj->Set(Nan::New("ps_recv").ToLocalChecked(), Nan::New<Integer>(ps.ps_recv));
+    stats_obj->Set(Nan::New("ps_drop").ToLocalChecked(), Nan::New<Integer>(ps.ps_drop));
+    stats_obj->Set(Nan::New("ps_ifdrop").ToLocalChecked(), Nan::New<Integer>(ps.ps_ifdrop));
     // ps_ifdrop may not be supported on this platform, but there's no good way to tell, is there?
 
-    NanReturnValue(stats_obj);
+    info.GetReturnValue().Set(stats_obj);
 }
 
-NAN_METHOD(PcapSession::Inject)
+void PcapSession::Inject(const Nan::FunctionCallbackInfo<Value>& info)
 {
-    NanScope();
+    Nan::HandleScope scope;
 
-    if (args.Length() != 1) {
-        NanThrowTypeError("Inject takes exactly one argument");
-        NanReturnUndefined();
+    if (info.Length() != 1) {
+        Nan::ThrowTypeError("Inject takes exactly one argument");
+        return;
     }
 
-    if (!node::Buffer::HasInstance(args[0])) {
-        NanThrowTypeError("First argument must be a buffer");
-        NanReturnUndefined();
+    if (!node::Buffer::HasInstance(info[0])) {
+        Nan::ThrowTypeError("First argument must be a buffer");
+        return;
     }
 
-    PcapSession* session = ObjectWrap::Unwrap<PcapSession>(args.This());
+    PcapSession* session = Nan::ObjectWrap::Unwrap<PcapSession>(info.Holder());
     char * bufferData = NULL;
     size_t bufferLength = 0;
-    Local<Object> buffer_obj = args[0]->ToObject();
+    Local<Object> buffer_obj = info[0]->ToObject();
     bufferData = node::Buffer::Data(buffer_obj);
     bufferLength = node::Buffer::Length(buffer_obj);
 
     if (pcap_inject(session->pcap_handle, bufferData, bufferLength) != (int)bufferLength) {
-        NanThrowError("Pcap inject failed.");
-        NanReturnUndefined();
+        Nan::ThrowError("Pcap inject failed.");
+        return;
     }
-    NanReturnUndefined();
+    return;
 }
-

--- a/pcap_session.h
+++ b/pcap_session.h
@@ -4,7 +4,7 @@
 #include <nan.h>
 #include <pcap/pcap.h>
 
-class PcapSession : public node::ObjectWrap {
+class PcapSession : public Nan::ObjectWrap {
 public:
     static void Init(v8::Handle<v8::Object> exports);
 
@@ -12,19 +12,19 @@ private:
     PcapSession();
     ~PcapSession();
 
-    static NAN_METHOD(New);
-    static _NAN_METHOD_RETURN_TYPE Open(bool live, _NAN_METHOD_ARGS);
-    static NAN_METHOD(OpenLive);
-    static NAN_METHOD(OpenOffline);
-    static NAN_METHOD(Dispatch);
-    static NAN_METHOD(Fileno);
-    static NAN_METHOD(Close);
-    static NAN_METHOD(Stats);
-    static NAN_METHOD(Inject);
+    static void New(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Open(bool live, const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void OpenLive(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void OpenOffline(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Dispatch(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Fileno(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Close(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Stats(const Nan::FunctionCallbackInfo<v8::Value>& info);
+    static void Inject(const Nan::FunctionCallbackInfo<v8::Value>& info);
     static void PacketReady(u_char *callback_p, const struct pcap_pkthdr* pkthdr, const u_char* packet);
 
-    v8::Persistent<v8::Function> packet_ready_cb;
-    static v8::Persistent<v8::Function> constructor;
+    Nan::Persistent<v8::Function> packet_ready_cb;
+    static Nan::Persistent<v8::Function> constructor;
 
     struct bpf_program fp;
     bpf_u_int32 mask;

